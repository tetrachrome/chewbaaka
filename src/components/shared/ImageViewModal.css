/**
 * ImageViewModal.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 29, 2020
 * Updated  : Aug 08, 2020
 */

div.ImageViewModalCoverTriggerOuterContainer {
  max-width: 100%;
  position: relative;
  background-size: cover;
  cursor: pointer;
}

div.ImageViewModalCoverTriggerOuterContainer:hover {
  opacity: 0.6;
}

div.ImageViewModalCoverTriggerImageIconInnerContainer {
  position: absolute;
  bottom: 8px;
  right: 0;
}

div.ImageViewModalComponentModalContentOuterContainer {
  margin: 0 auto;
}

div.ImageViewModalComponentModalContentImageContainer {
  margin: 0 auto;
}

img.ui.huge.image.ImageViewModalComponentModalContentImage {
  display: block;
  margin: 0 auto;
}
