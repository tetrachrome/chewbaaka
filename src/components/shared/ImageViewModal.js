/**
 * ImageViewModal.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 29, 2020
 * Updated  : Aug 07, 2020
 */

import React from 'react'

import { Icon, Image, Modal } from 'semantic-ui-react'

import { getElementStyleClassName } from '../../styling/styling'

import 'semantic-ui-css/semantic.min.css'

import './ImageViewModal.css'

if ( process.env.NODE_ENV === 'development' )
  require('./ImageViewModal-debug.css')

export default class ImageViewModal extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      isModalOpen: false
    }
    this.handleOpen = this.handleOpen.bind(this);
    this.handleClose = this.handleClose.bind(this);
  }

  handleOpen() {
    this.setState({
      isModalOpen: true
    });
  }

  handleClose() {
    this.setState({
      isModalOpen: false
    });
  }

  render() {
    const modalCoverTriggerContainerStyle = {
      width: this.props.width,
      height: this.props.height,
      backgroundImage: `url(${this.props.image})`,
    };

    return (
      <Modal
        trigger={
          <div
            className="ImageViewModalCoverTriggerOuterContainer"
            style={modalCoverTriggerContainerStyle}
            onClick={this.handleOpen}
            data-testid="ImageViewModalComponentCoverImageContainerTestId"
          >
            <div className="ImageViewModalCoverTriggerImageIconInnerContainer">
              <Icon
                name="file image"
                color="black"
                size="large"
                onClick={this.handleOpen}
              />
            </div>
          </div>
        }
        open={this.state.isModalOpen}
        onClose={this.handleClose}
        closeIcon
      >
        <Modal.Content>
          <div className={getElementStyleClassName("ImageViewModalComponentModalContentOuterContainer")}>
            <div className={getElementStyleClassName("ImageViewModalComponentModalContentImageContainer")}>
              <Image
                className="ImageViewModalComponentModalContentImage"
                size='huge'
                src={this.props.image}
              />
            </div>
            {/* <Modal.Description>
              <div className="ImageViewModalComponentModalDescriptionContainer">
              </div>
            </Modal.Description> */}
          </div>
        </Modal.Content>
      </Modal>
    );
  }
}
