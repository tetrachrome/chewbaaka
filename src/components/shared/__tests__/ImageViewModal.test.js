/**
 * ImageViewModal.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 29, 2020
 * Updated  : Jul 29, 2020
 */

import React from 'react'

import { render, screen } from '@testing-library/react'

import renderer from 'react-test-renderer'

import ImageViewModal from '../ImageViewModal'

import image from '../assets/cheetah-conservation-fund-logo.jpg'

test('renders ImageViewModal component', () => {
  render(
    <ImageViewModal
      width={720}
      height={480}
      image={image}
    />
  );

  const componentCoverImageContainerPart = screen.getByTestId("ImageViewModalComponentCoverImageContainerTestId");
  expect(componentCoverImageContainerPart).toBeInTheDocument();
});

test('ImageViewModal component snapshot', () => {
  const tree = renderer
    .create(
      <ImageViewModal
        width={720}
        height={480}
        image={image}
      />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
