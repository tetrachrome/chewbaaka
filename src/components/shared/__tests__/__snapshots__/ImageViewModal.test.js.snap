// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageViewModal component snapshot 1`] = `
<div
  className="ImageViewModalCoverTriggerOuterContainer"
  data-testid="ImageViewModalComponentCoverImageContainerTestId"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundImage": "url(cheetah-conservation-fund-logo.jpg)",
      "height": 480,
      "width": 720,
    }
  }
>
  <div
    className="ImageViewModalCoverTriggerImageIconInnerContainer"
  >
    <i
      aria-hidden="true"
      className="black file image large icon"
      onClick={[Function]}
    />
  </div>
</div>
`;
